### PR TITLE
feat/homepage

### PR DESCRIPTION
**Pull request by adith-a-danthi**
## Overview

- Add [tidle-ui](https://temp-dev--tidle-ui.netlify.app/)
- Add fontAwesome Icons
- Add ReactRouter
- Add Context + reducer for tasks
- Add Components
  - Navbar
  - Task Item
  - Modal
- Add Homepage

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -5,12 +5,18 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
-    <meta
-      name="description"
-      content="Web site created using create-react-app"
-    />
+    <meta name="description" content="Web site created using create-react-app" />
```
Comment (junaidshaikh-js) [inline public/index.html]: Add some description about your website

File: src/components/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+export default function Modal({ setShowModal, task = {}, editModal }) {
+  const [newTask, setNewTask] = useState({
+    title: '',
+    description: '',
+    duration: 15,
+    isCompleted: false,
+    ...task,
+  });
+  const { dispatchTasks } = useTasks();
```
Comment (junaidshaikh-js) [inline src/components/Modal.jsx:14]: Good that you used the name after the dispatch function to differentiate it from other dispatch functions from different contexts.

File: src/components/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  const submitHandler = () => {
+    editModal
+      ? dispatchTasks({ type: actionTypes.UPDATE_TASK, payload: newTask })
+      : dispatchTasks({ type: actionTypes.ADD_TASK, payload: { ...newTask, id: uuidV4() } });
+    setShowModal(false);
+  };
+
+  return (
+    <div className="modal-window is-active">
```
Comment (junaidshaikh-js) [inline src/components/Modal.jsx]: Use semantic group elements such as `section` or `article` instead of `div`

File: src/hooks/tasksReducer.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const tasksInitialState = [];
+
+const actionTypes = {
+  ADD_TASK: 'ADD_TASK',
+  REMOVE_TASK: 'REMOVE_TASK',
+  UPDATE_TASK: 'UPDATE_TASK',
+};
```
Comment (junaidshaikh-js) [inline src/hooks/tasksReducer.js]: The action-type object is used to just avoid some errors that may arise due to spelling mistakes in action types?

**Review by junaidshaikh-js — COMMENTED** [review]
Good folder structure. Clean code. 

Suggested changes can be applied. 

One thing about file structure why did you store `taskReducer` inside the `hooks` folder because it is not a hook and if it is a custom hook it should start with `use`

File: src/hooks/tasksReducer.js (hunk 4 above)
Comment (adith-a-danthi) [reply src/hooks/tasksReducer.js]: One scenario is spelling mistakes, the other is if you want to change the value of any of them then it ca be done in just one place

**Comment by adith-a-danthi** [comment]
> One thing about file structure why did you store `taskReducer` inside the `hooks` folder because it is not a hook and if it is a custom hook it should start with `use`

@junaidshaikh-js noted. Refactored the `taskReducer` to `reducers` folder

**Review by FarhanMobashir — APPROVED** [review]
-great code with proper naming and varaible
-good usage of small components

**Pull request merged**